src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import classNames from 'classnames';
+import {
+  useParams,
+  useSearchParams,
+} from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink/PersonLink';
+import { getFilteredPeople } from '../../utils/filterPeople';
+import { SearchParams } from '../../utils/searchHelper';
+import { SortBy } from '../../types/SortBy';
+import { SearchLink } from '../SearchLink';
+
+interface Props {
+  people: Person[],
+}
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+
+  const [searchParams] = useSearchParams();
+
+  const sortField = searchParams.get('sort') as keyof Person || '';
+  const sortOrder = searchParams.get('order') || '';
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const getSortIconClass = (sortBy: string): string => {
+    const sortByNormalized = sortBy.toLowerCase();
+
+    return classNames('fas', {
+      'fa-sort': sortField !== sortByNormalized,
+      'fa-sort-up': sortField === sortByNormalized && sortOrder === SortBy.Desc,
+      'fa-sort-down': sortField === sortByNormalized
+        && sortOrder === SortBy.Asc,
+    });
+  };
+
+  const getSortingParams = (sortBy: string): SearchParams => {
+    if (!sortOrder) {
+      return { order: SortBy.Asc, sort: sortBy };
+    }
+
+    if (sortOrder === SortBy.Asc) {
+      return { order: SortBy.Desc };
+    }
+
+    return { sort: null, order: null };
+  };
+
+  const peopleToRender = getFilteredPeople(
+    people,
+    sortField,
+    sortOrder,
+    sex,
+    query,
+    centuries,
+  );
+
+  const SORT_HEADERS = ['Name', 'Sex', 'Born', 'Died'];
```

Comment:
Such constants should be kept outside the creation of the component, so that they are not created anew during rerendering.